mm2src/lp_ordermatch.rs

```diff
@@ -3131,6 +3145,115 @@ pub async fn set_price(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, Strin
     Ok(try_s!(Response::builder().body(res)))
 }
 
+//update makerorder will update the previously created order with new volume and new price.
+pub async fn update_maker_order(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, String> {
+    let req: MakerOrderUpdateReq = try_s!(json::from_value(req));
+
+    let ordermatch_ctx = try_s!(OrdermatchContext::from_ctx(&ctx));
+    let mut my_orders = ordermatch_ctx.my_maker_orders.lock().await;
+
+    //Get the previous order details using uuid
+    let my_previous_order = match my_orders.get(&req.uuid) {
+        Some(O) => O,
+        None => return ERR!("Order with {} uuid is not found",&req.uuid),
+    };
+
+    let base_coin: MmCoinEnum = match try_s!(lp_coinfind(&ctx, &my_previous_order.base).await) {
+        Some(coin) => coin,
+        None => return ERR!("Base coin {} is not found", my_previous_order.base),
+    };
+
+    let rel_coin: MmCoinEnum = match try_s!(lp_coinfind(&ctx, &my_previous_order.rel).await) {
+        Some(coin) => coin,
+        None => return ERR!("Rel coin {} is not found", my_previous_order.rel),
+    };
+    if base_coin.wallet_only() {
+        return ERR!("Base coin is wallet only");
+    }
+    if rel_coin.wallet_only() {
+        return ERR!("Rel coin is wallet only");
+    }
+
+    // get the balance details of base-coin.
+    let my_balance = try_s!(base_coin.my_spendable_balance().compat().await);
+
+    let volume = if req.max {
+        let rel_coin_trade_fee = try_s!(
+            rel_coin
+                .get_receiver_trade_fee(FeeApproxStage::OrderIssue)
+                .compat()
+                .await
+        );
+        try_s!(check_other_coin_balance_for_swap(&ctx, &rel_coin, None, rel_coin_trade_fee).await);
+        try_s!(calc_max_maker_vol(&ctx, &base_coin,&my_balance, FeeApproxStage::OrderIssue).await)
+    } else {
+        try_s!(
+            check_balance_for_maker_swap(
+                &ctx,
+                &base_coin,
+                &rel_coin,
+                req.new_volume.clone(),
+                None,
+                None,
+                FeeApproxStage::OrderIssue
+            )
+            .await
+        );
+        req.new_volume
+    };
+
+
+    let conf_settings = OrderConfirmationsSettings {
+        base_confs: req.base_confs.unwrap_or_else(|| base_coin.required_confirmations()),
```

Comment:
The order confirmations settings are always overwritten by `base/rel_coin.required_confirmations()` this way. It is an error because initial settings might differ from `base/rel_coin.required_confirmations()`. Please consider changing these settings only when they are present in the request.